ex27/script.js

```diff
@@ -0,0 +1,50 @@
+let numeros = []
+
+function botao(){
+    document.getElementById("EscolhaNumero").style.display = "inline"
+}
+
+function Adicionar() {
+    let n = document.getElementById('Numero')
+    let numero = Number(n.value)
+    
+    if(n.value.trim() == ''){
+        alert('Informe um numero para adicionar a lista.')
+
+    } else {
+        if(!Repetido(numero)){
+            let lista = document.getElementById('Lista')
+            let adicionar = document.createElement('li')
+            adicionar.textContent = numero
+            lista.appendChild(adicionar)
+            
+            numeros.push(numero)
+        }
+    }
+    n.value = ''
+}
+
+function Repetido(numero) {
+    if(numeros.indexOf(numero) == -1){
+        return false
+
+    } else {
+        alert('Sua lista ja pussui este numero!')
+        return true 
+
+    }
+}
```

Comment:
como se entrar na primeira condicional (if) já vai dar um return, nunca entrara na segunda validacao (do else). Então poderia fazer assim

```suggestion
function Repetido(numero) {
    if(numeros.indexOf(numero) == -1){
        return false
    } 
    
    alert('Sua lista ja pussui este numero!')
    return true 
}
```

ainda, um outro jeito de verificar se tem o numero que vc quer é usar novamente as funçoes especificas de array. Existe uma (`.find`), que ele percorre o array até o encontrar o valor que voce deseja (se encontrar, retorna true, se nao, retorna false). Entao um outro jeito poderia ser:

```suggestion
function Repetido(numero) {
    if(numeros.find(n => n === numero)){
        alert('Sua lista ja pussui este numero!')
        return true 
    } 
    
    return false;
}
```

ainda, existe um outro jeito otimizado de fazer o if (que chamamos de operador ternário - https://pt.stackoverflow.com/questions/4907/como-funciona-este-if-else-com-e)

```suggestion
function Repetido(numero) {
    return numeros.find(n => n === numero) 
        ? (alert('Sua lista já possui este número!'), true) 
        : false;
}
```